src/SocialServer/Network/Send.cs

```diff
@@ -0,0 +1,225 @@
+		public static void SC_LOGIN_OK(ISocialConnection conn)
+		{
+			var packet = new Packet(Op.SC_LOGIN_OK);
+
+			conn.Send(packet);
+		}
+
+		/// <summary>
+		/// Dummy
+		/// </summary>
+		public static void SC_FROM_INTEGRATE(ISocialConnection conn, int type)
+		{
+			var packet = new Packet(Op.SC_FROM_INTEGRATE);
+
+			switch (type)
+			{
+				case 0x01:
+					packet.PutLong(conn.Account.Id);
+					packet.PutInt(0);
+					break;
+				case 0x14:
+					packet.PutLong(conn.Account.Id);
+					packet.PutInt(0); // 728
+					break;
+			}
+			conn.Send(packet);
+		}
+
+		/// <summary>
+		/// Login related
+		/// </summary>
+		/// <param name="conn"></param>
+		public static void SC_NORMAL_Unknown_00(ISocialConnection conn)
+		{
+			var packet = new Packet(Op.SC_NORMAL);
+			packet.PutInt(NormalOp.Social.Unknown_00);
+
+			conn.Send(packet);
+		}
+
+		/// <summary>
+		/// Login related
+		/// </summary>
+		/// <param name="conn"></param>
+		public static void SC_NORMAL_Unknown_02(ISocialConnection conn)
+		{
+			var packet = new Packet(Op.SC_NORMAL);
+			packet.PutInt(NormalOp.Social.Unknown_02);
+
+			conn.Send(packet);
+		}
+
+		public static void SC_NORMAL_Unknown_0C(ISocialConnection conn)
+		{
+			var packet = new Packet(Op.SC_NORMAL);
+			packet.PutInt(NormalOp.Social.Buff_0C);
+
+			packet.PutByte(0);
+			packet.PutLong(554643486671500);
+			packet.PutByte(1);
+			packet.PutLong(conn.Account.Id);
+			packet.PutInt(4723);
+			packet.PutInt(1);
+			packet.PutInt(0);
+			packet.PutByte(0);
+
+			conn.Send(packet);
+		}
+
+		/// <summary>
+		/// Dummy
+		/// </summary>
+		public static void SC_NORMAL_Unknown_19(ISocialConnection conn)
+		{
+			var packet = new Packet(Op.SC_NORMAL);
+
+			packet.PutInt(NormalOp.Social.Unknown_19);
+			packet.PutLong(conn.Account.Id);
+			packet.PutLong(conn.Account.Id);
+			packet.PutEmptyBin(16);
+			packet.PutLpString("WEEK");
+			packet.PutLong(1);
+
+			conn.Send(packet);
+		}
+
+		/// <summary>
+		/// Dummy
+		/// </summary>
+		public static void SC_NORMAL_Chat(ISocialConnection conn, Account account, long chatId, long l1, string targetTeamName, string message, byte b2)
+		{
+			var packet = new Packet(Op.SC_NORMAL);
+
+			packet.PutInt(NormalOp.Social.Chat);
+			packet.PutLong(chatId); // Chat Id
+			packet.PutLong(l1); // Chat Index?
+			packet.PutByte(1);
+			packet.PutDate(DateTime.Now);
+			packet.PutLpString(account.TeamName);
+			packet.PutShort(1001);
+			packet.PutLpString(message);
+			packet.PutByte(b2);
+			packet.PutInt(2);
+			packet.PutShort(1);
+			packet.PutByte(0);
+			packet.PutLpString(targetTeamName);
+			packet.PutLpString("GLOBAL");
+
+			conn.Send(packet);
+		}
+
+		/// <summary>
+		/// Dummy
+		/// </summary>
+		/// <param name="conn"></param>
+		/// <param name="account"></param>
+		/// <param name="message"></param>
+		public static void SC_NORMAL_CreateChatRoom(ISocialConnection conn, Account account, long chatId, string message = "!@#$NewRoomHasBeenCreated#@!")
```

Comment:
Another case of "we should probably use client messages"^^ Does it send the raw message if you don't wrap it in the lookup code or why did you decide to make the called include it?

Reply:
Actually, this opens up the question of whether we should even use client messages at all... We decided we wanted to handle the localization on the server-side, and client messages are the opposite. We're sending a string key here that gets turned into a localized message.

Reply:
So it's just my coding habit, when I have different parts of a packet, I use them as defaults, so when I test the packet, I change part at a time without having to rewrite all the parameters. I have changed it in newer commits.